server/src/main/java/org/opensearch/action/search/TransportCreatePitAction.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.search.SearchPhaseResult;
+import org.opensearch.search.internal.ShardSearchContextId;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+
+/**
+ * Transport action for creating PIT reader context
+ */
+public class TransportCreatePitAction extends HandledTransportAction<CreatePitRequest, CreatePitResponse> {
+
+    public static final String CREATE_PIT_ACTION = "create_pit";
+    private final TransportService transportService;
+    private final SearchTransportService searchTransportService;
+    private final ClusterService clusterService;
+    private final TransportSearchAction transportSearchAction;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+
+    @Inject
+    public TransportCreatePitAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        SearchTransportService searchTransportService,
+        ClusterService clusterService,
+        TransportSearchAction transportSearchAction,
+        NamedWriteableRegistry namedWriteableRegistry
+    ) {
+        super(CreatePitAction.NAME, transportService, actionFilters, in -> new CreatePitRequest(in));
+        this.transportService = transportService;
+        this.searchTransportService = searchTransportService;
+        this.clusterService = clusterService;
+        this.transportSearchAction = transportSearchAction;
+        this.namedWriteableRegistry = namedWriteableRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, CreatePitRequest request, ActionListener<CreatePitResponse> listener) {
+        Thread t = new Thread(() -> {
+            CreatePitController controller = new CreatePitController(
+                request,
+                searchTransportService,
+                clusterService,
+                transportSearchAction,
+                namedWriteableRegistry,
+                task,
+                listener
+            );
+            controller.execute();
+        });
+        t.start();
+    }
```

Comment:
Let's avoid separate thread to create a PIT since we aren't doing something which is compute heavy on network threads. Instead would suggest calling the CreatePitController#executeCreatePit directly on the same caller thread